torax/sources/external_current_source.py

```diff
@@ -228,17 +232,30 @@ def jext_hires(
     """Return the external current density profile along the hires cell grid."""
     assert isinstance(dynamic_source_runtime_params, DynamicRuntimeParams)
     self.check_mode(dynamic_source_runtime_params.mode)
-    return source.get_source_profiles(
-        dynamic_runtime_params_slice=dynamic_runtime_params_slice,
-        dynamic_source_runtime_params=dynamic_source_runtime_params,
-        geo=geo,
-        core_profiles=None,
-        # There is no model for this source.
-        model_func=(lambda _0, _1, _2, _3: jnp.zeros_like(geo.r_hires_norm)),
-        formula=self.hires_formula,
-        output_shape=geo.r_hires_norm.shape,
+
+    # Interpolate prescribed values onto the hires grid
+    hires_prescribed_values = jnp.where(
+      dynamic_source_runtime_params.mode == runtime_params_lib.Mode.PRESCRIBED,
```

Comment:
Check against the value of the enum, not the enum itself, like shown here: https://github.com/google-deepmind/torax/blob/bb08d5a432ad94235042d7b107c8e6314b00389d/torax/sources/source.py#L471

Python will end up comparing an enum value vs an int and always return false.

Reply:
Resolved in 95ae88f699bac31839c0529f4529a3c2dfdd7801